Extension/src/LanguageServer/settingsPanel.ts

```diff
@@ -0,0 +1,224 @@
+    cppStandard: "cppStandard"
+};
+
+export interface ViewStateEvent {
+    isActive: boolean;
+}
+
+export class SettingsPanel {
+    private configValues: config.Configuration;
+    private configDirty: boolean = false;
+    private settingsPanelViewStateChanged = new vscode.EventEmitter<ViewStateEvent>(); 
+    private panel: vscode.WebviewPanel;
+    private disposable: vscode.Disposable = undefined;
+    private disposablesPanel: vscode.Disposable = undefined;
+    private static readonly viewType: string = 'settingsPanel';
+    private static readonly title: string = 'C/C++ Configurations';
+
+    constructor() {
+        this.configValues = { name: undefined };
+        this.disposable = vscode.Disposable.from(
+            this.settingsPanelViewStateChanged,
+            vscode.window.onDidChangeWindowState(this.onWindowStateChanged, this)
+        );
+    }
+
+    public createOrShow(activeConfiguration: config.Configuration): void {
+        const column: vscode.ViewColumn = vscode.window.activeTextEditor
+                ? vscode.window.activeTextEditor.viewColumn
+                : undefined;
+
+        // Show existing panel
+        if (this.panel) {
+            this.panel.reveal(column, false);
+            return;
+        }
+
+        // Create new panel
+        this.panel = vscode.window.createWebviewPanel(
+            SettingsPanel.viewType,
+            SettingsPanel.title,
+            column || vscode.ViewColumn.One,
+            {
+                enableCommandUris: true,
+                enableScripts: true,
+
+                // Restrict the webview to only loading content from these directories
+                localResourceRoots: [
+                    vscode.Uri.file(util.extensionContext.extensionPath), 
+                    vscode.Uri.file(path.join(util.extensionContext.extensionPath, 'ui')), 
+                    vscode.Uri.file(path.join(util.extensionContext.extensionPath, 'out', 'ui'))]
+            }
+        );
+
+        this.panel.iconPath = vscode.Uri.file(util.getExtensionFilePath("LanguageCCPP_color_128x.png"));
+
+        this.disposablesPanel = vscode.Disposable.from(
+            this.panel,
+            this.panel.onDidDispose(this.onPanelDisposed, this),
+            this.panel.onDidChangeViewState(this.onViewStateChanged, this),
+            this.panel.webview.onDidReceiveMessage(this.onMessageReceived, this)
+        );
+
+        this.panel.webview.html = this.getHtml();
+
+        this.updateWebview(activeConfiguration);
+    }
+
+    public get SettingsPanelViewStateChanged(): vscode.Event<ViewStateEvent> { 
+        return this.settingsPanelViewStateChanged.event;
+    }
+
+    public getLastValuesFromConfigUI(): config.Configuration {
+        return this.configValues;
+    }
+
+    public updateConfigUI(configuration: config.Configuration): void {
+        if (this.panel) {
+            this.updateWebview(configuration);
+        }
+    }
+
+    //TODO: validate input paths

```

Comment:
Did you want to keep this?

Reply:
Yes, keeping some of the code to validate inputs as I'm planning to work on them after insiders.